### ISSUE TITLE
jsonencode does not encode backslashes

### ISSUE THREAD

**Issue by voroniys** [hashicorp/terraform#20931]
### Terraform Version

```
Terraform v0.11.11
```

### Terraform Configuration Files

```hcl
${jsonencode(file("script.rb"))}
...
```

### Expected Behavior
The file script.rb is multiline file that contains also sequences \n, for instance 
puts "something\n"
Expected that these sequences preserved as is.

### Actual Behavior
In incoded JSON all these sequences stay as \n, so when decoded back naturally they all gets converted to the newline characters.

### Steps to Reproduce
Just use jsonecode on anything that contains \n sequences

**Comment by apparentlymart** [comment]
Sorry for this strange behavior @voroniys, and thanks for reporting it.

Fortunately the `jsonencode` function was rewritten from scratch to work with the new language changes in the forthcoming Terraform v0.12.0 release, and the new version of it doesn't seem to have this bug:

```
> file("foo.txt")
hello\nworld

> jsonencode(file("foo.txt"))
"hello\\nworld\n"
> jsondecode(jsonencode(file("foo.txt")))
hello\nworld
```

That fix is already in the master branch ready for release. If you'd like to verify it, you could try with [v0.12.0-beta1](https://www.hashicorp.com/blog/announcing-terraform-0-1-2-beta1), though please don't use that pre-release against production infrastructure.

The v0.11.x series is now frozen aside from some limited changes being driven by the Terraform Enterprise team, so unfortunately we won't be able to push a fix for this in a v0.11.x release, and I don't have a ready workaround for this issue.

Because this is already fixed in the `master` branch, I'm going to close this out. Thanks again for reporting it!

**Comment by voroniys** [comment]
To my mind this reaction is fully incorrect. 

The version 12 is not yet released, and even relese date is not yet known. In such situation statement that we are not going to fix any bugs in version we are currently distributiung as mainline is an uitmost form of disregards to your users comunity. Especially with the fact, that you already have a solution. 

I'll never believe that backporting of function from 12 back to 11 and releasing new version will take for the good developer more than 1 hour. But instead your are saing - we don't care that all of you will have a problem with this function till v12 is released and no workaround for it, you all not worth of 1 hour of my time.
